flink-tests/src/test/java/org/apache/flink/test/checkpointing/SnapshotCompatibilityITCase.java

```diff
@@ -0,0 +1,199 @@
+		JobClient jobClient = submitJobInitially(env(startAligned, 0, emptyMap()));
+		Thread.sleep(FIRST_RUN_EL_COUNT * FIRST_RUN_BACKPRESSURE_MS); // wait for all tasks to run and some backpressure from sink
+		Future<Map<String, Object>> accFuture = jobClient.getAccumulators(getClass().getClassLoader());
+		Future<String> savepointFuture = jobClient.stopWithSavepoint(false, folder.toURI().toString());
+		return new Tuple2<>(savepointFuture.get(), accFuture.get());
+	}
+
+	private Tuple2<String, Map<String, Object>> runAndTakeExternalCheckpoint() throws Exception {
+		JobClient jobClient = submitJobInitially(externalCheckpointEnv(startAligned, folder, 100));
+		File metadata = waitForChild(waitForChild(waitForChild(folder))); // structure: root/attempt/checkpoint/_metadata
+		cancelJob(jobClient);
+		return new Tuple2<>(metadata.getParentFile().toString(), emptyMap());
+	}
+
+	private static JobClient submitJobInitially(StreamExecutionEnvironment env) throws Exception {
+		return env.executeAsync(dag(FIRST_RUN_EL_COUNT, true, FIRST_RUN_BACKPRESSURE_MS, env));
+	}
+
+	private Map<String, Object> runFromSavepoint(String path, boolean isAligned, int totalCount) throws Exception {
+		StreamExecutionEnvironment env = env(isAligned, 50, Collections.singletonMap(SAVEPOINT_PATH, path));
+		return env.execute(dag(totalCount, false, 0, env)).getJobExecutionResult().getAllAccumulatorResults();
+	}
+
+	@SuppressWarnings({"OptionalGetWithoutIsPresent", "ConstantConditions"})
+	private static File waitForChild(File dir) throws InterruptedException {
+		while (dir.listFiles().length == 0) {
+			Thread.sleep(50);
+		}
+		return Arrays.stream(dir.listFiles()).max(Comparator.naturalOrder()).get();
+	}
+
+	private void cancelJob(JobClient jobClient) throws InterruptedException, java.util.concurrent.ExecutionException {
+		jobClient.cancel().get();
+		try {
+			jobClient.getJobExecutionResult(getClass().getClassLoader()); // wait for cancellation
+		} catch (Exception e) {
+			// ignore cancellation exception
+		}
+	}
+
+	private StreamExecutionEnvironment externalCheckpointEnv(boolean isAligned, File dir, int checkpointingInterval) {
+		Map<ConfigOption<?>, String> cfg = new HashMap<>();
+		cfg.put(CHECKPOINTS_DIRECTORY, dir.toURI().toString());
+		cfg.put(MAX_RETAINED_CHECKPOINTS, Integer.toString(Integer.MAX_VALUE)); // prevent deletion of checkpoint files while it's being checked and used
+		StreamExecutionEnvironment env = env(isAligned, checkpointingInterval, cfg);
+		env.getCheckpointConfig().enableExternalizedCheckpoints(RETAIN_ON_CANCELLATION);
+		return env;
+	}
+
+	@SuppressWarnings("unchecked")
+	private StreamExecutionEnvironment env(boolean isAligned, int checkpointingInterval, Map<ConfigOption<?>, String> cfg) {
+		Configuration configuration = new Configuration();
+		for (Map.Entry<ConfigOption<?>, String> e: cfg.entrySet()) {
+			configuration.setString((ConfigOption<String>) e.getKey(), e.getValue());
+		}
```

Comment:
Why not pass `configuration` directly instead of `cfg`?

Reply:
For some cases constructing a `Map` requires less boilerplate.